java/arcs/core/storage/StoreInterface.kt

```diff
@@ -25,5 +26,6 @@ interface IStore<Data : CrdtData, Op : CrdtOperation, ConsumerData> {
 data class StoreOptions(
     val storageKey: StorageKey,
     val type: Type,
-    val versionToken: String? = null
+    val versionToken: String? = null,
+    val coroutineScope: CoroutineScope? = null
```

Comment:
Please do not add coroutineScope here. This is for values related to the data stored in the store.

Reply:
Ah I thought it's just providing options or configurations for a Store.

Reply:
If you make it a second parameter for the store constructors, that should be ok.

Reply:
A second parameter needs to plumb the parameter to multiple critical interfaces/classes/paths like activationFactory, DirectStore, ReferenceModeStore, etc.
Given you are refactoring storage stack, interfaces and protocols, I'll leave it as it is.

Reply:
Isn't StoreOptions sent trough parcelables? Did you update ParcelableStoreOption? I don't really know how you'd Parcel a CoroutineScope

Reply:
ParcelableStoreOptions reads and writes fields from a StoreOptions object.
I don't read / write the `coroutineScope` field so ParcelableStoreOptions works as it did before.

The coroutineScope will only be used on the service end specifically for ReferenceModeStore and DirectStore.
Client side doesn't communicate the coroutineScope with service.

Reply:
Btw jibbl is working on refactoring storage stack.
So the scope plumbing will be changed later.